stravalib/model.py

```diff
@@ -81,40 +116,60 @@ def __repr__(self):
             attrs.append('id={0}'.format(self.id))
         if hasattr(self.__class__, 'name'):
             attrs.append('name={0!r}'.format(self.name))
-        if hasattr(self.__class__, 'resource_state') and self.resource_state is not None:
+        if (
+            hasattr(self.__class__, 'resource_state')
+            and self.resource_state is not None
+        ):
             attrs.append('resource_state={0}'.format(self.resource_state))
 
-        return '<{0}{1}>'.format(self.__class__.__name__, ' ' + ' '.join(attrs) if attrs else '')
+        return '<{0}{1}>'.format(
+            self.__class__.__name__, ' ' + ' '.join(attrs) if attrs else ''
+        )
 
 
 class ResourceStateEntity(BaseEntity):
     """
     Mixin for entities that include the resource_state attribute.
     """
-    resource_state = Attribute(int, (META, SUMMARY, DETAILED))  #: The detail-level for this entity.
+
+    resource_state = Attribute(
+        int, (META, SUMMARY, DETAILED)
+    )  #: The detail-level for this entity.
 
 
 class IdentifiableEntity(ResourceStateEntity):
     """
     Mixin for entities that include an ID attribute.
     """
-    id = Attribute(int, (META, SUMMARY, DETAILED))  #: The numeric ID for this entity.
+
+    id = Attribute(
+        int, (META, SUMMARY, DETAILED)
+    )  #: INT The numeric ID for this entity.
 
 
 class BoundEntity(BaseEntity):
     """
-    Base class for entities that support lazy loading additional data using a bound client.
+    Base class for entities that support lazy loading additional data using a 
+    bound client.
+
+    Attributes
+    ----------
+    bind_client : stravalib.client.Client
```

Comment:
So here  i suspect that bind_client is also a class because it's a Client object. 
Soi need to remove it from being an attribute. i'm just really getting confused by the complexity of the various objects called within objects. in some moduleslike exc we have classes that are this simpe

```
class LoginFailed(AuthError):
    pass
```

<img width="764" alt="Screen Shot 2022-12-13 at 8 44 07 AM" src="https://user-images.githubusercontent.com/7649194/207378651-5bd75951-0798-4954-bc87-4861cf19242f.png">